xds/src/main/java/io/grpc/xds/CdsLoadBalancer2.java

```diff
@@ -185,7 +185,7 @@ private void handleClusterDiscovered() {
       }
       LoadBalancerProvider lbProvider = null;
       Object lbConfig = null;
-      if (root.result.lbPolicy() == LbPolicy.RING_HASH) {
+      if (root.result != null && root.result.lbPolicy() == LbPolicy.RING_HASH) {
```

Comment:
If root.result is null, then because of the above code, the queue is immediately emptied and `instances` is also empty, and because  `instances` is empty, the code won't reach here.